sql/src/main/java/io/crate/metadata/Functions.java

```diff
@@ -169,13 +193,51 @@ private FunctionImplementation getBuiltin(FunctionName functionName, List<DataTy
     private FunctionImplementation getBuiltinByArgs(FunctionName functionName,
                                                     List<? extends FuncArg> argumentsTypes,
                                                     SearchPath searchPath) {
+        // V2
+        FunctionImplementation impl = resolveFunctionBySignature(
+            functionName,
+            Lists2.map(argumentsTypes, FuncArg::valueType),
+            searchPath
+        );
+        if (impl != null) {
+            return impl;
+        }
+
         FunctionResolver resolver = lookupFunctionResolver(functionName, searchPath, functionResolvers::get);
         if (resolver == null) {
             return null;
         }
         return resolveFunctionForArgumentTypes(argumentsTypes, resolver);
     }
 
+    @Nullable
+    private FunctionImplementation resolveFunctionBySignature(FunctionName name,
+                                                              List<DataType> arguments,
+                                                              SearchPath searchPath) {
+        var candidates = functionImplementations.get(name);
+        if (candidates == null && name.schema() == null) {
+            for (String pathSchema : searchPath) {
+                FunctionName searchPathFunctionName = new FunctionName(pathSchema, name.name());
+                candidates = functionImplementations.get(searchPathFunctionName);
+                if (candidates != null) {
+                    break;
+                }
+            }
+        }
+        if (candidates != null) {
+            for (FuncResolver candidate : candidates) {
+                Signature boundSignature = new SignatureBinder(candidate.getSignature(), true)
+                    .bind(arguments);
+                if (boundSignature != null) {
+                    // TODO: check for more specific function
```

Comment:
Can this TODO be resolved?

Reply:
I'll follow up on here later, maybe even on a dedicated commit.